frontend/src/app/settings/core/user-settings/user-settings.component.ts

```diff
@@ -38,50 +48,51 @@ export class UserSettingsComponent implements OnInit {
   }
 
   get username(): FormControl {
-    return this.createAdministratorFormGroup.get('username') as FormControl;
+    return this.createUserFormGroup.get('username') as FormControl;
   }
 
-  createAdministrator() {
-    if (this.createAdministratorFormGroup.valid) {
-      this.userService
-        .updateRoleOfUser(
-          this.createAdministratorFormGroup.value.username as string,
-          'administrator'
-        )
-        .subscribe({
-          next: () => {
-            this.toastService.showSuccess(
-              'Role of user updated',
-              this.createAdministratorFormGroup.value.username +
-                ' has now the role administrator'
-            );
-            this.getUsers();
-          },
-          error: () => {
-            this.toastService.showError(
-              'Update of role failed',
-              'The role of ' +
-                this.createAdministratorFormGroup.value.username +
-                ' was not updated'
-            );
-          },
-        });
+  userAlreadyExistsValidator(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      if (this.users) {
+        for (const user of this.users) {
+          if (user.name == control.value) {
+            return { userAlreadyExists: true };
+          }
+        }
+      }
+      return null;
```

Comment:
```suggestion
      const existingUser = this.users.find(
        (user) => user.name == control.value
      );

      if (existingUser) {
        return { userAlreadyExists: true };
      }
      return null;
```
I think since we initialize the users with an empty list, it can't be undefined, so we don't necessarily need the outer if. And by using "Array.find" we can further reduce the nesting depth.

Reply:
I get:`ERROR Error: Uncaught (in promise): TypeError: this.users is not iterable` when removing the `if` statement. But I applied your second suggestion.

Reply:
It seems that the error occurs because the user array is initialized after the form creation.  If you put the `users: User[] = [];` before the initialization of the form, it works fine. So it might be a good idea to always define the form directly above the constructor, but again, this is more of a general discussion than part of this PR.

Reply:
Didn't see this open conversation before merge, but I fixed it in a new PR: https://github.com/DSD-DBS/capella-collab-manager/pull/325